messagix/client.go

```diff
@@ -388,3 +393,28 @@ func (c *Client) GetTaskId() int {
 	c.activeTasks = append(c.activeTasks, id)
 	return id
 }
+
+func (c *Client) EnableSendingMessages() {
+	c.sendMessagesCond.L.Lock()
+	c.canSendMessages = true
+	c.sendMessagesCond.Broadcast()
+	c.sendMessagesCond.L.Unlock()
+}
+
+func (c *Client) WaitUntilCanSendMessages(timeout time.Duration) error {
+	c.sendMessagesCond.L.Lock()
+	defer c.sendMessagesCond.L.Unlock()
+
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+
+	for !c.canSendMessages {
+		select {
+		case <-timer.C:
+			return fmt.Errorf("timeout waiting for sending messages")
+		default:
+			c.sendMessagesCond.Wait()
+		}
+	}
+	return nil
+}
```

Comment:
I think this would solve the two issues mentioned above (the blocking forever and the deadlock), wdyt @tulir?
```suggestion
func (c *Client) WaitUntilCanSendMessages(timeout time.Duration) error {
	timer := time.NewTimer(timeout)
	defer timer.Stop()

	done := make(chan struct{})
	go func() {
		c.sendMessagesCond.L.Lock()
		defer c.sendMessagesCond.L.Unlock()
		c.sendMessagesCond.Wait()
		close(done)
	}()

	for !c.canSendMessages {
		select {
		case <-timer.C:
			return fmt.Errorf("timeout waiting for canSendMessages")
		case <-done:
			if c.canSendMessages {
				return nil
			}
		}
	}
	return nil
}
```